mfs/dir.go

```diff
@@ -363,6 +363,18 @@ func (d *Directory) AddChild(name string, nd ipld.Node) error {
 		return err
 	}
 
+	//TODO: Resolve where to manage a possible transition to sharding.
```

Comment:
This is ugly, and it's not equivalent to transitioning inside `unix.io.Directory.AddChild`.